### PR TITLE
Taavirehemagi android widget. Closes #493 #494 #28

### LINKED ISSUE #493: Recent entrys list to widget
Add three latest time entrys based on the start time.
Epic #28

### PR DESCRIPTION
**Pull request by taavirehemagi**
Lock screen widget for Android.

Closes #493 #494 #28

### CODE REVIEW

File: Joey/Properties/AndroidManifest.xml [hunk 1]
```diff
@@ -24,5 +25,21 @@
...
+                <action android:name="android.appwidget.action.APPWIDGET_UPDATE" />
+            </intent-filter>
+            <meta-data
+                android:name="android.appwidget.provider"
+                android:resource="@xml/widget_info" />
+        </receiver>
+        <service
+            android:name="toggl.joey.net.WidgetListViewService"
+            android:permission="android.permission.BIND_REMOTEVIEWS"
+            android:exported="false" />
```
Comment (monday8am) [inline Joey/Properties/AndroidManifest.xml]: Avoid to modify directly the AndroidManifest.xml. 
The info related with the widget is duplicated. Try to put all this info using this way:
```
[BroadcastReceiver]
[IntentFilter (new string [] { "android.appwidget.action.APPWIDGET_UPDATE" })]
[MetaData ("android.appwidget.provider", Resource = "@xml/widget_info")]
```

File: Phoebe/Data/Views/WidgetDataView.cs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                return activeTimeEntry;
+            }
+        }
+
+        public bool HasRunning
+        {
+            get {
+                return hasRunning;
+            }
+        }
```
Comment (monday8am) [inline Phoebe/Data/Views/WidgetDataView.cs]: HasRunning? I think IsRunning is more clear.

**Comment by monday8am** [comment]
Joey/Resources/drawable/widget_color_shape.xml
Joey/Resources/layout/keyguard_widget.axml
Joey/Resources/layout/widget_list_item.axml
Joey/Resources/layout/widget_login.axml
Joey/Resources/xml/widget_info.xml

this files don´t follow the previous name convention: WidgetInfo, WidgetLogin, etc.

**Comment by taavirehemagi** [comment]
I know, It's because CamelCase layout files are not accessible from widget.

**Comment by monday8am** [comment]
- There is an extra 1 number in all times showed by widget.
- There is a difference between the two times showed.

screen shot 2015-02-11 at 11 43 33

**Comment by monday8am** [comment]
In general, I don´t like the idea of checking from widget to app every second. It should be (IMHO) in the opposite way, every time exist an update, the app refresh the widget. I´ll research a bit and try to suggest you a solution.

**Comment by monday8am** [comment]
superseded by #547

**Pull request closed without merging**